### ISSUE TITLE
run_custom_training silently skips training when script defines def train() without calling it

### ISSUE THREAD

**Issue by abhijeet-dhumal** [kubeflow/mcp-server#19]
### Description

_make_train_func unconditionally wraps the script body inside a generated def train():. When the user's script already defines a top-level def train(): without calling it - the natural way to write a named training function - the inner function is defined inside the wrapper but never called. The TrainJob completes with exit code 0 and no training output.

### Steps to Reproduce

```
run_custom_training(
    script="""
def train():
    import torch
    print("Training started")
    for epoch in range(3):
        print(f"Epoch {epoch}")
""",
    gpu_per_node=0,
    confirmed=True,
)
```

1. Job reaches Complete status
2. get_training_logs returns only pip install output — no "Training started" or epoch lines

Root Cause
kubeflow_mcp/trainer/api/training.py,
```
lines = script.strip().splitlines()
wrapped = f"def {func_name}():\n"
for line in lines:
    wrapped += f"    {line}\n"  # blindly indents everything
```
The Kubeflow Trainer SDK (backends/kubernetes/utils.py, get_command_using_train_func) calls inspect.getsource(train_func) on the returned callable and then appends a train() call at module level. The final script on the cluster becomes:

```
def train():        # outer wrapper - what the SDK calls
    def train():    # user's function - defined, never called
        print("Training started")
        for epoch in range(3):
            print(f"Epoch {epoch}")
train()             # appended by SDK - runs outer, outer silently exits

```

### Expected Behavior

_make_train_func detects the top-level def train(): and returns it directly. The SDK invokes the user's function. Training output appears in logs.

## Fix 
Use ast.parse to detect a top-level def train (regex would false-positive on string literals). If func_args is given and the user's signature doesn't include the required params, raise a clear ValueError rather than falling back to a wrapping that also silently fails.

### Version

master

### Python Version

3.12

**Comment by abhijeet-dhumal** [comment]
@sjhddh @algojogacor @kaladin-Stormborn I see you all have raised same bug fixes for same issue 
FYI : The ideal way to work on any issue is to assign yourself that issue first with /assign tag to make sure noone else is repeating same work 
Please take a look at CONTRIBUTING.md guide

**Comment by abhijeet-dhumal** [comment]
/assign @Kaladin-Stormborn

I have reviewed all 3 fixes and most closer one to the expectation i found out is #23 , so assigning accordingly.. Thanks!

**Comment by reckless-sherixx** [comment]
@abhijeet-dhumal is the issue resolved or i can work on this issue

**Comment by abhijeet-dhumal** [comment]
> [@abhijeet-dhumal](https://github.com/abhijeet-dhumal) is the issue resolved or i can work on this issue

You are the first who added correct label and assigned officially 😄 .. please proceed!
you can take inspiration from existing PRs.. 

If someone is already working on this.. please give heads up here
cc: @Kaladin-Stormborn

**Comment by reckless-sherixx** [comment]
sure i will take inspirations from the previous prs and I will start working on this issue

**Comment by reckless-sherixx** [comment]
@abhijeet-dhumal  I have raised a pr for this issue . Kindly review it and let me know if any changes required .
